src/ctia/lib/es/index.clj

```diff
@@ -76,3 +92,63 @@
   [conn index-name mappings]
   (when-not ((index-exists?-fn conn) conn index-name)
     ((index-create-fn conn) conn index-name :mappings mappings)))
+
+(defn create-alias!
+  "create an index alias simple or filtered"
+  ([conn index alias]
+   ((update-alias-fn conn)
+    conn
+    {:add {:index index
+           :alias alias}}))
+  ([conn index alias routing filter]
+   ((update-alias-fn conn)
+    conn
+    {:add {:index index
+           :alias alias
+           :routing routing
+           :filter filter}})))
+
+(s/defn create-aliased-index!
+  "create an index with an alias for a slice"
+  [state :- ESConnState
+   index-name :- s/Str]
+
+  (create!
+   (:conn state)
+   index-name
+   (:mapping state))
+
+  (create-alias!
+   (:conn state)
+   index-name
+   (:index state)))
+
+(s/defn create-filtered-alias!
+  "create a filtered index alias"
+  [state :- ESConnState
+   name :- s/Str
+   routing :- s/Str
+   filter :- {s/Any s/Any}]
+
+  (create!
+   (:conn state)
+   (:index state)
+   (:mapping state))
+
+  (create-alias!
+   (:conn state)
+   (:index state)
+   name
+   routing
+   filter))
+
+(def cached-create-filtered-alias!
+  (memo/ttl create-filtered-alias!
+            :ttl/threshold
+            alias-create-cache-ttl-ms))
+
+(def cached-create-aliased-index!
```

Comment:
I think you should use a FIFO memo instead of a TTL memo.  Dates will change as time progresses, so older entries should fall out.  I think the way that TTL memo is implemented, it only purges entries when they are hit.  Since we won't try to index in old slices (once a new time interval starts), the TTL cache could grow in size over time.  FIFO on the other hand fits our pattern and won't grow beyond a specified size.